src/task/MonitorTask.ts

```diff
@@ -438,4 +462,42 @@ export default class MonitorTask
       poorConnectionCount: this.context.poorConnectionCount,
     };
   };
+
+  /**
+   * Degrade video codec to an alternative codec
+   */
+  private degradeVideoCodec(cause: string): void {
+    // Degrade video codec if there are other codec options and current codec is not H264 CBP or VP8
+    if (
+      this.context.meetingSupportedVideoSendCodecPreferences !== undefined &&
+      this.context.meetingSupportedVideoSendCodecPreferences.length > 1 &&
+      !(
+        this.context.meetingSupportedVideoSendCodecPreferences[0].equals(
+          VideoCodecCapability.h264ConstrainedBaselineProfile()
+        ) ||
+        this.context.meetingSupportedVideoSendCodecPreferences[0].equals(VideoCodecCapability.vp8())
+      )
+    ) {
+      const newMeetingSupportedVideoSendCodecPreferences: VideoCodecCapability[] = [];
+      for (const capability of this.context.videoSendCodecPreferences) {
+        if (!capability.equals(this.context.meetingSupportedVideoSendCodecPreferences[0])) {
+          newMeetingSupportedVideoSendCodecPreferences.push(capability);
+        }
+      }
+      if (newMeetingSupportedVideoSendCodecPreferences.length > 0) {
+        this.context.logger.info(
+          `Downgrading codec from ${this.context.meetingSupportedVideoSendCodecPreferences[0].codecName} to ${newMeetingSupportedVideoSendCodecPreferences[0].codecName} due to ${cause}`
```

Comment:
Seem like we just downgrade from [0] to the new codec [0]? Would that always be in [1] of the previous meetingSupportedVideoSendCodecPreferences?

Reply:
Yes, it should be the case. Still deriving the list there to avoid things unexpected.